src/api/environment.cc

```diff
@@ -615,8 +617,9 @@ MaybeLocal<Object> GetPerContextExports(Local<Context> context) {
 
   Local<Object> exports = Object::New(isolate);
   if (context->Global()->SetPrivate(context, key, exports).IsNothing() ||
-      InitializePrimordials(context).IsNothing())
+      InitializePrimordials(context, isolate_data).IsNothing()) {
```

Comment:
If `GetPerContextExports` is called with `nullptr` isolate data, it could be better to CHECK primordials are already initialized to prevent unexpected results.